### PR TITLE
✨ Support collection in gift page

### PR DESCRIPTION
**Pull request by williamchong**

### CODE REVIEW

File: src/pages/nft/gift/index.vue [hunk 1]
```diff
@@ -4,7 +4,15 @@
...
-            localeLocation({ name: 'nft-class-classId', params: { classId } })
+            collectionId
+              ? localeLocation({
+                  name: 'nft-collection-collectionId',
+                  params: { collectionId },
+                })
+              : localeLocation({
+                  name: 'nft-class-classId',
+                  params: { classId },
+                })
```
Comment (nwingt) [inline src/pages/nft/gift/index.vue:15]: Can be put this in `nftOrCollectionMixin` and reuse it in `handleClickViewClass`?

File: src/pages/nft/gift/index.vue [hunk 2]
```diff
@@ -91,6 +104,12 @@ export default {
     classId() {
       return this.$route.query.class_id;
     },
+    collectionId() {
+      return this.$route.query.collection_id;
+    },
+    primaryKey() {
+      return this.collectionId || this.classId;
+    },
```
Comment (nwingt) [inline src/pages/nft/gift/index.vue:112]: This should put in `nftOrCollectionMixin`?

File: src/pages/nft/gift/index.vue (hunk 1 above)
Comment (williamchong) [reply src/pages/nft/gift/index.vue]: added but not sure how to name this

File: src/pages/nft/gift/index.vue (hunk 2 above)
Comment (williamchong) [reply src/pages/nft/gift/index.vue]: yes in fact it already exists in there...

File: src/pages/nft/gift/index.vue (hunk 1 above)
Comment (nwingt) [reply src/pages/nft/gift/index.vue]: I think is fine for now

**Review by nwingt — APPROVED** [review]

**Pull request merged**